categorize_adjectives/kmedoids.py

```diff
@@ -0,0 +1,67 @@
+"""Categorize Adjectives using K-medoids algorithm
+Here we only try to categorize most frequent 100 adjectives
+"""
+import os
+import random
+import pickle
+import numpy as np
+from pyclustering.cluster.kmedoids import kmedoids
+from pyclustering.utils.metric import distance_metric, type_metric
+from libraries.bert import BertSimilarity
+
+# Folder where generated files are stored
+generated_file = os.path.join(os.getcwd(), '..', 'generated_files/')
+
+# Initializing the BERT Similarity function and also storing calculated distances in array.
+BERT = BertSimilarity()
+dis_array = np.zeros((150, 150)) - 1
+
+def distance(w_1, w_2):
+    """
+    Calulates and Returns the distance between w1 and w2
+    Also stores the distance in dis_array
+    """
+    if dis_array[w_1, w_2] == -1:
+        dis_array[w_1, w_2] = 1-(BERT.get_similarity([(selected_adj[int(w_1)], selected_adj[int(w_2)])],
+                                 layer=-2)[0])
+        dis_array[w_2, w_1] = dis_array[w_1, w_2]
+    return dis_array[w_1, w_2]
+
+# Fetching the noun to adj dictionary from pickle file
+with open(generated_file + 'noun_to_adj_score.dat', 'wb') as f:
+    noun_to_adj = pickle.load(f)
+
```

Comment:
Please refactor the below code to create multiple functions and add a main function.
Same as above file.